tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -1476,76 +1474,78 @@ public static void test_prepare_insert() throws Exception {
 
 	public static void test_read_only() throws Exception {
 		Path database_file = Files.createTempFile("duckdb-jdbc-test-", ".duckdb");
-		database_file.toFile().delete();
+		Files.deleteIfExists(database_file);
 
-		String jdbc_url = "jdbc:duckdb:" + database_file.toString();
+		String jdbc_url = "jdbc:duckdb:" + database_file;
 		Properties ro_prop = new Properties();
 		ro_prop.setProperty("duckdb.read_only", "true");
 
 		Connection conn_rw = DriverManager.getConnection(jdbc_url);
 		assertFalse(conn_rw.isReadOnly());
+		assertFalse(conn_rw.getMetaData().isReadOnly());
 		Statement stmt = conn_rw.createStatement();
 		stmt.execute("CREATE TABLE test (i INTEGER)");
 		stmt.execute("INSERT INTO test VALUES (42)");
 		stmt.close();
-		// we cannot create other connections, be it read-write or read-only right now
-		try {
-			Connection conn2 = DriverManager.getConnection(jdbc_url);
-			fail();
-		} catch (Exception e) {
-		}
 
-		try {
-			Connection conn2 = DriverManager.getConnection(jdbc_url, ro_prop);
-			fail();
-		} catch (Exception e) {
-		}
+		// Verify we can open additional write connections
+		// Using the Driver
+		try (Connection conn = DriverManager.getConnection(jdbc_url);
+				 Statement stmt1 = conn.createStatement();
+				 ResultSet rs1 = stmt1.executeQuery("SELECT * FROM test")) {
+			rs1.next();
+			assertEquals(rs1.getInt(1), 42);
+		}
+		// Using the direct API
+		try (Connection conn = ((DuckDBConnection) conn_rw).duplicate();
+				 Statement stmt1 = conn.createStatement();
+				 ResultSet rs1 = stmt1.executeQuery("SELECT * FROM test")) {
+			rs1.next();
+			assertEquals(rs1.getInt(1), 42);
+		}
+
+		// At this time, mixing read and write connections on Windows doesn't work
+		// Read-only when we already have a read-write
+//		try (Connection conn = DriverManager.getConnection(jdbc_url, ro_prop);
+//				 Statement stmt1 = conn.createStatement();
+//				 ResultSet rs1 = stmt1.executeQuery("SELECT * FROM test")) {
+//			rs1.next();
+//			assertEquals(rs1.getInt(1), 42);
+//		}
 
-		// hard shutdown to not have to wait on gc
-		DuckDBDatabase db = ((DuckDBConnection) conn_rw).getDatabase();
 		conn_rw.close();
-		db.shutdown();
```

Comment:
Closing the last connection closes the database. We have a separate test verifying this behavior.